recipe/meta.yaml

```
@@ -0,0 +1,191 @@
+  # test/torchaudio_unittest/backend/dispatcher/smoke_test.py::test_wav->run_smoke_test->get_save_func(): on s390x the
+  # function get_save_func() is unable to instantiate a StreamWriter through the ffmpeg save_audio function
+  # Call stack:
+  # get_save_func() --> save() --> backend.save(...) --> torchaudio/_backend/ffmpeg.py::save_audio() --> s.open() -->
+  # torchaudio/io/_stream_writer.py::self._s.open(option)   throws a RuntimeError
+  # RuntimeError: Failed to write header:
+  # /tmp/tmpx2on2_qz/torchaudio_unittest.backend.dispatcher.smoke_test.SmokeTest.test_wav/test.wav (Function not implemented)
+  #
+  # since the save function constitutes one of the basic functionalities of torchaudio, it is best to skip the platform altogether
+  # as long as other backends (sox, libsndfile) are not available for s390x
+  skip: True  # [s390x]
+  script:
+    - export USE_FFMPEG=ON                         # [not win]
+    - set USE_FFMPEG=ON                            # [win]
+    - export BUILD_SOX=OFF                         # [not win]
+    - set BUILD_SOX=OFF                            # [win]
+    # On linux, Torchaudio's build system will find gnu openmp preferentially to other openmp implementations,
+    # and we can't keep gnu openmp out of the host environment due to issues with our openmp packages.
+    # That could lead to conflicts with other packages. Turn it off for linux.
+    - export USE_OPENMP=OFF                        # [linux]
+    - export USE_OPENMP=ON                         # [osx]
+    - set USE_OPENMP=ON                            # [win]
+    - export BUILD_TORCHAUDIO_PYTHON_EXTENSION=ON  # [not win]
+    - set BUILD_TORCHAUDIO_PYTHON_EXTENSION=ON     # [win]
+    - export FFMPEG_ROOT=${PREFIX}                 # [not win]
+    - set FFMPEG_ROOT=%LIBRARY_PREFIX%             # [win]
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  missing_dso_whitelist:
+    - "**/libc10.so"                # [linux]
+    - "**/libtorch_cpu.so"          # [linux]
+    - "**/libtorch_python.so"       # [linux]
+    - "**/libtorchaudio_ffmpeg.so"  # [linux or osx]
+    - "**/libtorchaudio.so"         # [linux or osx]
+    - "**/libc10.dylib"             # [osx]
+    - "**/libtorch_cpu.dylib"       # [osx]
+    - "**/libtorch_python.dylib"    # [osx]
+    - "**/torch_cpu.dll"            # [win]
+    - "**/c10.dll"                  # [win]
+    - "**/torch_python.dll"         # [win]
+    - "**/libtorchaudio_ffmpeg.pyd" # [win]
+    - "**/libtorchaudio.pyd"        # [win]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+    - {{ compiler('c') }}
+    - cmake
+    - make      # [not win]
+    - patchelf  # [linux]
+  host:
+    - python
+    - setuptools
+    - wheel
+    - pip
+    - pytorch 2.1.0
+    - pybind11 2.12.0
+    - ffmpeg 6.1.1
+    - llvm-openmp  # [osx]
+  run:
+    - python
+    - {{ pin_compatible('ffmpeg') }}
```

```suggestion
    - ffmpeg
```
won't make a difference in practice but I prefer to rely on the run_export pinnning

---

I thought it did make a difference because `{{ pin_compatible('ffmpeg') }}` should ultimately resolve to `ffmpeg >=6.0.0,<7.0.0`

---

so does the run_export, or am I missing something? https://github.com/AnacondaRecipes/ffmpeg-feedstock/blob/4ec654b3c316c484b0d1247ffcad2f2eecba903b/recipe/meta.yaml#L24

---

it will actually resolve to `ffmpeg >=6.1.1, <7.0.0` see [here](https://docs.conda.io/projects/conda-build/en/latest/resources/variants.html#pinning-at-the-recipe-level). And the behaviour for `pin_compatible` and `pin_subpackage` seems to be one and the same as far as I can tell